### DISCUSSION TITLE
How to initialize a task without first run?

### DISCUSSION THREAD

**Discussion by hwebb** [go-co-op/gocron#176] [Q&A] [answered]
For example I have the task set like: `s.Every(60).Minutes().SingletonMode().Do(task)`
I'll probably have many of the above launching at different intervals, some takes a long time to run. I initialize all of them at launch of the script, but at the moment I launch the script, all of the tasks run at once a first time, then after that initial run it all goes to place, but I would like to avoid running everything at once as it could be resource intensive at that moment and would run things I don't necessarily want to run right now.
Am I doing it wrong?

**Comment by JohnRoesler** [comment, marked as answer]
Hi @hwebb you can set the individual job or the entire scheduler to wait for the first interval before running rather than running immediately.

```go
s.Every(60).Minutes().WaitForSchedule().Do(task)

or

s.WaitForScheduleAll()
s.Every(60).Minutes().Do(task)
```

The `SinglettonMode()` is relevant when you have long running tasks that may run longer than the scheduled interval. 

References:
https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.WaitForSchedule
https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.WaitForScheduleAll
https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.SingletonMode

**Reply by hwebb** [reply]
Thanks, I wonder how I missed that.

**Reply by JohnRoesler** [reply]
If you any suggestions on how to make it more clear - I would appreciate that! I want the documentation to be easy to follow!
